python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -291,6 +292,12 @@ def _create_repository_using_definitions_args(
         sensors_with_resources,
     ) = _attach_resources_to_jobs_and_instigator_jobs(jobs, schedules, sensors, resource_defs)
 
+    asset_specs = [obj for obj in (assets or []) if isinstance(obj, AssetSpec)]
+    assets_with_specs_replaced = [obj for obj in (assets or []) if not isinstance(obj, AssetSpec)]
+    if asset_specs:
+        with disable_dagster_warnings():
+            assets_with_specs_replaced.append(AssetsDefinition(specs=asset_specs))
+
```

Comment:
a little helper function that produces a variable called `canonicalized_assets_defs` might be more clear here.

Reply:
I suspect such a helper would also be useful at other entry points? E.g. `materialize?

Reply:
`materialize` just lives on top of `Definitions`, so doesn't need it, but seems like a sensible change anyway

Reply:
👍 refactored along these lines - much nicer this way.